tools/c7n_azure/c7n_azure/resources/postgresql_server.py

```diff
@@ -69,3 +71,69 @@ def _query_rules(self, resource):
                 continue
             resource_rules.add(rule)
         return resource_rules
+
+
+@PostgresqlServer.filter_registry.register('configuration-parameter')
+class ConfigurationParametersFilter(ValueFilter):
+    """Filter by configuration parameter for this postresql server
+
+    Configurations are made available to the filter as a map with each
+    key holding the name of the configuration and each value holding
+    the properties of the Configuration as defined here:
+    https://learn.microsoft.com/en-us/python/api/azure-mgmt-rdbms/azure.mgmt.rdbms.postgresql.models.configuration?view=azure-python
+
+    :example:
+
+    Example JSON document showing the data format provided to the filter
+
+    .. code-block:: json
+
+      {
+        "value": "off",
+        "description": "Logs each successful connection.",
+        "defaultValue": "on",
+        "dataType": "Boolean",
+        "allowedValues": "on,off",
+        "source": "user-override",
+        "isConfigPendingRestart": "False",
+        "isDynamicConfig": "True"
+      }
+
+    :example:
+
+    Find Postgresql servers with log_connections not enabled
+
+    .. code-block:: yaml
+
+        policies:
+          - name: sql-database-no-log-connections
+            resource: azure.postgresql-server
+            filters:
+              - type: configuration-parameter
+                name: log_connections
+                key: value
+                op: ne
+                value: 'on'
+
+    """
+
+    schema = type_schema(
+        'configuration-parameter',
+        required=['type', 'name'],
+        rinherit=ValueFilter.schema,
+        name=dict(type='string')
+    )
+
+    def __call__(self, resource):
+        key = f'c7n:config-params:{self.data["name"]}'
+        if key not in resource['properties']:
+            client = self.manager.get_client()
```

Comment:
let's move the client instantiation outside the for loop

Reply:
@thisisshi Do you mean to move it outside of `__call__`?

Reply:
oh sorry, i misread this and was thinking of a different PR feel free to ignore my comment here